src/pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -245,7 +245,7 @@ def _restore_quantization_callbacks(self) -> None:
             if state:
                 # The Quantization callbacks have a special method that must be called before restoring the weights
                 # of the model
-                callback._load_before_model(self.trainer.model, deepcopy(state))
+                callback._load_before_model(self.trainer.lightning_module, deepcopy(state))
```

Comment:
Can someone with a bit more knowledge around quantization take a look whether it's correct? I suggested it because it seemed to make the most sense (and we were expecting `LightningModule` along the way anyway, but I am not 100% sure. cc @carmocca @awaelchli

Reply:
Yes it is correct, because _restore_quantization_callbacks gets called before the model gets wrapped in the strategy, so at this point in time, we have: `self.trainer.lightning_module == self.trainer.model`.